In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [10]:
from google.colab import files 
uploaded=files.upload()

Saving emisiones2.csv to emisiones2.csv


In [11]:
data= pd.read_csv("emisiones2.csv",delimiter =";",header=0, skipfooter=59) ##Elimina las ultimas filas del archivo
data

/usr/local/lib/python3.7/dist-packages/pandas/util/_decorators.py:311: ParserWarning:

Falling back to the 'python' engine because the 'c' engine does not support skipfooter; you can avoid this warning by specifying engine='python'.



Año Periodo    Carbón  Petróleo      Gas  Cemento  Combustión   Otros
0    1750     Año      9.35       NaN      NaN      NaN         NaN     NaN
1    1751     Año      9.35       NaN      NaN      NaN         NaN     NaN
2    1752     Año      9.35       NaN      NaN      NaN         NaN     NaN
3    1753     Año      9.35       NaN      NaN      NaN         NaN     NaN
4    1754     Año      9.36       NaN      NaN      NaN         NaN     NaN
..    ...     ...       ...       ...      ...      ...         ...     ...
266  2016     Año  14360.63  11959.22  6967.33  1478.76      387.69  111.80
267  2017     Año  14453.70  12130.80  7117.01  1500.49      422.86  113.15
268  2018     Año  14718.15  12190.05  7457.57  1566.74      412.22  115.12
269  2019     Año  14573.22  12229.64  7553.39  1608.47      434.60  115.12
270  2020     Año  13976.10  11072.50  7399.51  1626.37      435.03     NaN

[271 rows x 8 columns]

In [12]:
import plotly.express as px

In [40]:
fig=px.line(data, x="Año", y= data.columns[2:],
            color_discrete_map={'Carbón':'mediumorchid','Petróleo':'teal','Gas':'tomato', 
                                'Cemento':'limegreen','Combustión':'yellow','Otros':'magenta'}, 
            title='Evolución en las emisiones globales de CO2 por origen')


fig.update_xaxes(title_text='Año')
fig.update_yaxes(title_text='Toneladas de CO2 (Millones)')
fig.update_layout(width= 1000, height=500)
fig.write_html("data.html")
fig.show()

## Efectos de la revolución industrial en las emisiones de CO2.

Las cifras de Global Carbon Project recogen un aumento exponencial de las emisiones de CO2 por combustibles fósiles desde la época preindustrial, hasta nuestros dias; donde el mundo pasó de producir 9.35 millones de toneladas en el año 1.750 a 34.000 millones de toneladas de CO2 en el año 2020, al rededor del mundo. 

El gráfico evidencia que el Carbón ha sido el producto que por excelencia ha movido el mundo a lo largo de la historia y por ende generado más emisiones de CO2. Este mineral fue usado principalmente en China e Inglaterra como la principal fuente de generación de combustible. 

Asimismo, en siglo XVIII (1700-1800) se descubrieron en Norteamérica grandes yacimientos de carbón y comenzó a ser usado como el combustible que alimentaba a la máquina de vapor. Por su parte en el siglo XIX se consolidó la generalización de la extracción y consumo de carbón utilizado para el desarrollo industrial de los países y el ferrocarril. 

Con la Segunda Revolución Industrial (a mediados y finales del siglo XIX) se genera un punto de inflexión en el consumo energético y surgieron el petróleo y el gas como fuentes energéticas. La industria petrolera empieza en 1859 con el primer pozo estadounidense, inaugurado por el coronel Edwin Drake en Oil Creek, en Pensilvania, a partir de entonces su consumo ha crecido de forma exponencial, mucho más rápido que el gas, hasta convertirse en la fuente de energía más utilizada a mediados del siglo XX.

El consumo del gas natural creció fuertemente con la crisis del 2008-2009, periodo en el que los precios del petróleo y sus derivados alcanzaron los niveles más altos y el gas surgió como una fuente de energia alternativa, con mejor rendimiento, reduciendo costes y emisiones de CO2.

En la actualidad, aunque se esten tratando de utilizar tecnologías más limpias y eficientes, la producción de Carbón es aproximadamente 2,6 veces mayor que la del petróleo y 1,9 veces mayor que la de gas natural. Su uso se mantine gracias a los países avanzados que cuentan grandes reservas del mineral, al ser seguro, confiable, fácil de almacenar, y disponible y ademas de los multiples usos del carbón; generación de electricidad, producción de acero, fabricación de cemento, producción de combustibles líquidos, entre otros, lo mantienen en el primer lugar de los minerales consumidos a nivel mundial y por ende con mayores emisiones de CO2. 
